### PR TITLE
[SWA-210][INFRA] - Update pipelines and monorepo build process

### PR DESCRIPTION
**Pull request by ElRodrigote**
## Fixes: [SWA-210](https://linear.app/swaprhq/issue/STK-210/update-pipelines-and-build-process)

# Description
* Update `build` command to generate SDK package needed files.

# Visual Evidence 
## `bun run build` command at root folder
image

## pipeline build output
image

# How to test the changes

1) Pull this branch
2) In the root folder:
  * `bun run clean:deep` to remove all the Git ignored files and directories
  * Run `bun install` to get all the project dependencies
  * Run `bun run build` to build all the packages in the monorepo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,13 +12,14 @@
...
   "scripts": {
-    "build": "turbo run build",
+    "build": "cd packages/sdk && yarn typechain && cd ../.. && turbo run build lint",
     "build:app": "cd packages/sdk && yarn typechain && turbo run build --filter=@stackly/sdk --filter=app",
     "build:landing": "cd packages/sdk && yarn typechain && turbo run build --filter=@stackly/sdk --filter=landing",
     "dev": "turbo run dev",
     "start": "turbo run start",
     "lint": "turbo run lint",
     "clean": "turbo run clean",
+    "clean:deep": "git clean -fdx",
```
Comment (berteotti) [inline package.json:22]: Why this git command? Show we add git commands to `package.json`?

File: package.json [hunk 2]
```diff
@@ -12,13 +12,14 @@
     "packages/*"
   ],
   "scripts": {
-    "build": "turbo run build",
+    "build": "cd packages/sdk && yarn typechain && cd ../.. && turbo run build lint",
     "build:app": "cd packages/sdk && yarn typechain && turbo run build --filter=@stackly/sdk --filter=app",
     "build:landing": "cd packages/sdk && yarn typechain && turbo run build --filter=@stackly/sdk --filter=landing",
```
Comment (berteotti) [inline package.json:17]:
```suggestion
    "build": "cd packages/sdk && bun typechain && cd ../.. && turbo run build lint",
    "build:app": "cd packages/sdk && bun typechain && turbo run build --filter=@stackly/sdk --filter=app",
    "build:landing": "cd packages/sdk && bun typechain && turbo run build --filter=@stackly/sdk --filter=landing",
```

Let's change it to bun!

File: package.json (hunk 1 above)
Comment (ElRodrigote) [reply package.json:22]: I left a brief explanation in the PR comment:
* Create a new `clean:depp` command useful to rebuild from scratch. It removes:
  * All `untracked` files
  * `-f` to force the deletion
  * `-d` to remove untracked directories
  * `-x` but only files and directories ignored by Git (present in `.gitignore`).
  
I'm not 100% sure about placing this here, but wanted to know what you guys feel about it. I think it's a useful command (that I didn't know about) and wanted to share it but I don't mind if we don't have it here. Feels more like a local command alias than a package command TBH.

Comment (ElRodrigote) [reply package.json]: Yeah, totally, removing now

**Review by berteotti — APPROVED** [review]
LGTM!

**Pull request merged**